error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/StreamRules.java

```diff
@@ -379,4 +380,16 @@ boolean after(Stream<T> stream) {
       return stream.allMatch(e -> test(e));
     }
   }
+
+  static final class StreamMapReduceIntSum<T> {
+    @BeforeTemplate
+    int before(Stream<T> stream, Function<? super T, Integer> mapper) {
+      return stream.map(mapper).reduce(0, Integer::sum);
+    }
+
+    @AfterTemplate
+    int after(Stream<T> stream, ToIntFunction<T> mapper) {
```

Comment:
This rule assumes that the `mapper` matched by the `@BeforeTemplate` is a lambda expression or method reference, rather than a reference to an actual `Function<? super T, Integer>` instance. To guard against this we could introduce a new `Matcher` and pass it to a `@Matches` annotation. The alternative is to defer this work and add an `XXX` comment. Let us know if you're up for the former :)

Reply:
Sure, I'm up for it! I just need to take some time and figure out how to do it :)

Reply:
Cool! A good place to start is to check out [these classes](https://github.com/PicnicSupermarket/error-prone-support/tree/master/refaster-support/src/main/java/tech/picnic/errorprone/refaster/matchers) :)